src/main/java/com/develhope/spring/vehicles/service/VehicleService.java

```diff
@@ -47,5 +49,14 @@ public boolean existsById(Long id) {
         return vehicleRepository.existsById(id);
     }
 
+    public Collection<MostPurchasedModel> mostPurchasedVehicles () {
+          return vehicleRepository.findMostPurchasedVehicleModel();
```

Comment:
non stai usando un DTO, dovresti integrarlo ed aggiungere la la mappatura per MostPusrchasedModel, nel pratico non so come si faccia per un'interfaccia ma Lorenzo ha detto che tutti i controller devono lavorare solo con DTO per non esporre alcuna classe del programma

Reply:
Non l'ho convertito a dto perchè questo a me sembra già un dto e infatti Lorenzo ha detto che nei fatti è così, quindi non sapevo se in effetti servisse fare tutto il lavoro di conversione per mostPurchasedModel